python/cudf/cudf/_lib/spillable_buffer.pxd

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+
+from libc.stdint cimport uintptr_t
+from libcpp.memory cimport shared_ptr
+from libcpp.vector cimport vector
+
+
+cdef class SpillLock:
+    cdef vector[shared_ptr[void]] _expose_counters
+
+    cdef add(self, shared_ptr[void] expose_counter)
+
+
+cdef class SpillableBuffer:
+    cdef object __weakref__
+    cdef shared_ptr[void] _expose_counter
+    cdef object _lock
+    cdef bint _exposed
```

Comment:
I think we need better names between `_expose_counter` and `exposed` to show that one is _internal_ (a counter of `Column` objects) while the other is _external_ (in general, since typically it'll be for something outside cudf that we request the pointer even though in theory we could call `ptr` internally).

Reply:
Removed `_expose_counter`, we now use a `self._spill_locks = weakref.WeakSet()` to maintain known spill locks